src/coreComponents/mesh/generators/HexCellBlockManager.cpp

```diff
@@ -0,0 +1,1126 @@
+
+    // Maybe the oriented facets would be useful - if they are the ones recomputed 
+    // later one by the FaceManager
+    faceToNodes[curFace][0] = cB.getElementNode(cellInBlock, Hex::facetVertex[faceToStore][0]);
+    faceToNodes[curFace][1] = cB.getElementNode(cellInBlock, Hex::facetVertex[faceToStore][1]);
+    faceToNodes[curFace][2] = cB.getElementNode(cellInBlock, Hex::facetVertex[faceToStore][2]);
+    faceToNodes[curFace][3] = cB.getElementNode(cellInBlock, Hex::facetVertex[faceToStore][3]);
+  }
+  return true;
+}
+
+
+bool HexMeshConnectivityBuilder::computeNodesToFaces( ArrayOfSets<localIndex> & nodeToFaces ) const
+{
+  ArrayOfArrays<localIndex> faceToNodes; 
+  computeFacesToNodes( faceToNodes );
+
+  // 1 - Counting  - Quite unecessary for hexahedral meshes 
+  //( 12 for regular nodes  - TODO check a good max for singular node )
+  std::vector<unsigned int> nbFacesPerNode(nbNodes, 0);
+  for(unsigned int i = 0; i < faceToNodes.size(); ++i)
+  {
+    nbFacesPerNode[faceToNodes[i][0]]++;
+    nbFacesPerNode[faceToNodes[i][1]]++;
+    nbFacesPerNode[faceToNodes[i][2]]++;
+    nbFacesPerNode[faceToNodes[i][3]]++;
+  }
+  localIndex valuesToReserve = std::accumulate(nbFacesPerNode.begin(), nbFacesPerNode.end(), 0);
+
+  // 2 - Allocating 
+  nodeToFaces.resize(0);
+  nodeToFaces.resize(nbNodes);
+  nodeToFaces.reserveValues( valuesToReserve );
+  for (localIndex n = 0; n < nbNodes; ++n)
+  {
+    nodeToFaces.appendSet( nbFacesPerNode[n] );
+  }
+  
+  // 3 - Filling 
+  for(unsigned int i = 0; i < faceToNodes.size(); ++i)
+  {
+    nodeToFaces.insertIntoSet( faceToNodes[i][0],i );
+    nodeToFaces.insertIntoSet( faceToNodes[i][1],i );
+    nodeToFaces.insertIntoSet( faceToNodes[i][2],i );
+    nodeToFaces.insertIntoSet( faceToNodes[i][3],i );
+  }
+
+  return false;
+}
+
+
+// TODO We have a problem - where are the Element index valid ?
+bool HexMeshConnectivityBuilder::computeFacesToElements( array2d<localIndex> & faceToElements ) const
+{
+  faceToElements.resize(0);
+  faceToElements.resize(nbFaces, 2);
+  // Initialize contents to NO_ID 
+  faceToElements.setValues< serialPolicy >( NO_ID );
+
+  for(unsigned int curFace = 0; curFace < m_uniqueFaces.size(); ++curFace)
+  {
+    localIndex f = m_uniqueFaces[curFace];
+    LocalFaceIndex neighbor = m_allFacesToNeighbors[f];
+
+    LocalCellIndex c = f/Hex::nbFacets;
+    LocalCellIndex cNeighbor = -1;
+    
+    localIndex cellInBlock = getBlockCellFromManagerCell(c).second;
+    localIndex neighborCellInMaybeNotSameBlock = -1;
+    
+    if (neighbor != -1) {
+      cNeighbor = neighbor/Hex::nbFacets;
+      neighborCellInMaybeNotSameBlock = getBlockCellFromManagerCell( cNeighbor ).second;
+    }
+
+    faceToElements[curFace][0] = cellInBlock;
+    faceToElements[curFace][1] = neighborCellInMaybeNotSameBlock;
+  }
+  return true;
+}
+
+// Using a set means bad performances and annoying code 
+// TODO Change set to vector
+void HexMeshConnectivityBuilder::getFacesAroundEdge( 
```

Comment:
What do you mean by around? The faces containing this edge? This could be some kind of `edge to face`?

Reply:
Around is indeed  vague. I mean incident, and yes the faces containing this edge. 
Name changed to getOneEdgeToFaces